From f1a847da164a00eb98b61089139689243f2f591d Mon Sep 17 00:00:00 2001
From: Peng Fan <peng.fan@nxp.com>
Date: Fri, 9 Nov 2018 14:13:57 +0800
Subject: [PATCH 5071/5242] MLK-20302-1 ARM64: dts: imx8qm: add pl031 node

commit  022e1590f3da337cacd579924686da22ca49c3fa from
https://source.codeaurora.org/external/imx/linux-imx.git

Add rtc pl031 and a dummy clock node.

The pl031 is emulated by XEN, the address and interrupt is fixed in XEN.

Signed-off-by: Peng Fan <peng.fan@nxp.com>
Reviewed-by: Flynn xu <flynn.xu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8qm-mek-domu.dts     |   20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-domu.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-domu.dts
index daccc8a..b0dc5b3 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-domu.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-domu.dts
@@ -170,6 +170,26 @@
 		interrupt-parent = <&gic>;
 	};
 
+	clocks {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		clk0: clock@0 {
+		        compatible = "fixed-clock";
+		        reg = <0>;
+		        #clock-cells = <0>;
+		        clock-frequency = <24000000>;
+		};
+	};
+
+	rtc0: rtc@23000000 {
+		compatible = "arm,pl031", "arm,primecell";
+		reg = <0x0 0x23000000 0x0 0x1000>;
+		interrupts = <GIC_SPI 32 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&clk0>;
+		clock-names = "apb_pclk";
+	};
+
 	passthrough {
 		compatible = "simple-bus";
 		ranges;
-- 
1.7.9.5

